spyder_kernels/console/tests/test_console_kernel.py

```diff
@@ -388,13 +405,32 @@ def test_runcell(tmpdir):
         p = tmpdir.join("cell-test.py")
         p.write(code)
 
-        # Attach cell_code to the IPython shell instance to simulate
-        # that the code was sent from Spyder's Editor
-        client.execute(u"get_ipython().cell_code = '{}'".format(code))
-        client.get_shell_msg(block=True, timeout=TIMEOUT)
+        class Signal():
+            def connect(self, function):
+                self.function = function
+
+        # Fake Qt signal
+        iopub_recieved = Signal()
+        client.iopub_channel.message_received = iopub_recieved
+        # Open comm
+        comm_manager = CommManager(client)
+        kernel_comm = KernelComm()
+        kernel_comm._register_comm(comm_manager.new_comm('spyder_api'))
+
+        def process_msg():
+            msg = {'msg_type': None}
+            while msg['msg_type'] != 'comm_msg':
+                msg = client.get_iopub_msg(block=True, timeout=TIMEOUT)
```

Comment:
Why does this need to be a blocking call now?

Reply:
I am not sure I understand how this could not be blocking, I need to finish the test before the end of the function?